packages/build-info/src/get-framework.ts

```diff
@@ -14,6 +14,9 @@ export async function getFramework(frameworkId: FrameworkName, project: Project)
     if (detected) {
       return detected
     }
+
+    // this indicate that framework's custom "detect" method doesn't honor the forced framework
+    throw new Error(`Forced framework "${frameworkId}" was not detected.`)
```

Comment:
This just change error being thrown - previously it would throw
> Invalid framework "remix". It should be one of: analog, angular, assemble, astro, blitz, brunch, cecil, create-react-app, docpad, docusaurus, eleventy, ember, expo, gatsby, gridsome, grunt, gulp, harp, hexo, hugo, hydrogen, jekyll, metalsmith, middleman, next, nuxt, observable, parcel, phenomic, quasar, qwik, react-static, redwoodjs, remix, roots, sapper, solid-js, solid-start, stencil, svelte, svelte-kit, vite, vue, vuepress, wintersmith, wmr, zola

which make no sense, so this just adds specific error message for what is actually happening

Reply:
It should never be visible to user because I did add test in `get-framework.test.ts` that checks if we ever hit that scenario - I'm testing it for all the frameworks we have, force the framework and the project looks like empty directory - this won't fully guarantee that it will never happen as only empty directory case is tested, but it should prevent most likely cases from being unnoticed and thus block the merge of changes that would lead to hitting this error case.

The difficulty with providing something actionable to user is that only custom code in each framework `.detect()` method can cause problems